### ISSUE TITLE
filter does not use functools.wraps

### ISSUE THREAD

**Issue by belongwqz** [kata198/func_timeout#2]
```
def a():
    pass

@func_set_timeout(xx)
def b():
    pass

print(a.__name__, b.__name__)
"a, <lambda>" # wrong, it should be "a, b"
```

**Comment by kata198** [comment]
I agree

submit a patch if you'd like, else I'll get to it sometime in the next few days

Thanks

**Comment by kata198** [comment]
**RESOLVED**

Thank you for your contribution.

I went with a slightly different implementation I did here: https://github.com/kata198/func_timeout/commit/00685ec5465670085538734e9fcf369093d51e1a 

And included a unit test (fails before change, passes after) which tests every distinct combination of the _func\_timeout_ decorator.

Since decorators are just functions, my implementation calls "wraps(func)" which returns a decorator which will insert the correct meta info, and we explicitly call that returned function on our decorator, like so:

    return wraps(func)(_function_wrapper)

I chose this instead of rewriting the code to be able to use the decorator syntax as there is a particular issue involving StoppableThread wherein you could get the wrong traceback (sometimes doubled-up) in a certain set of circumstances. The way the decorators are written now resolves that issue. It's kind of hairy because python doesn't officially support stopping threads at all, the way func_timeout implements it is unique (we raise an exception in the python context such that the next python lib call in C will terminate execution with a special BaseException type). Since this is not a supported operation (though this implementation DOES WORK), it is subject to some weirdness.

So in summary, your commit is not wrong, and I credited you in both the ChangeLog and commit message, but I used a slightly different implementation to retain the careful balance that is StoppableThread and exceptions.

I've pushed this as version 4.3.0, released to pypi and github, and updated the archlinux AUR packages. I'm going to close this issue.

**Comment by belongwqz** [comment]
@kata198 awesome
